### PR TITLE
Regenerate AzureCommunicationChat

### PR DESCRIPTION
**Pull request by tjprescott**
Necessary for release of AzureSDK.

### CODE REVIEW

File: sdk/communication/AzureCommunicationChat/Source/AzureCommunicationChatClient.swift [hunk 1]
```diff
@@ -16,19 +16,24 @@ import Foundation
...
-        case v20200921preview2 = "2020-09-21-preview2"
+        /// API version ""
+        case v = ""
 
         /// The most recent API version of the
         public static var latest: ApiVersion {
-            return .v20200921preview2
+            return .v
         }
     }
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationChat/Source/AzureCommunicationChatClient.swift:38]: @sacheu looks like there's still an issue here with how Autorest.Swift is (or rather, is not) generating the api version field.

File: sdk/communication/AzureCommunicationChat/Package.swift [hunk 2]
```diff
@@ -24,7 +24,7 @@ let package = Package(
     dependencies: [
         .package(
             url: "https://github.com/Azure/azure-sdk-for-ios.git",
-            .branch("master")
+            .revision("3e3c80d60173613c8dd4cb6b219188cf5070e8e7")
```
Comment (bsiegel) [inline sdk/communication/AzureCommunicationChat/Package.swift:27]: Is this change intended to be merged?

Comment (tjprescott) [reply sdk/communication/AzureCommunicationChat/Package.swift:27]: Yes, though we may need to change it to something else like the release tag? I'm curious how either can really even matter since if you download AzureSDK, AzureCore is baked in.

File: sdk/communication/AzureCommunicationChat/Source/AzureCommunicationChatClient.swift (hunk 1 above)
Comment (tjprescott) [reply sdk/communication/AzureCommunicationChat/Source/AzureCommunicationChatClient.swift]: I have manually reverted this and updated the issue in the Autorest.Swift repo to fix this.

**Comment by tjprescott** [comment]
There were numerous issues with the generated code that we have documented here (https://github.com/Azure/autorest.swift/issues/258) but this is now ready to merge and release. Many thanks to @sacheu for testing the SDK.

**Review by bsiegel — APPROVED** [review]
The manual fixes look reasonable to me. If the tests are passing, :shipit:
